kip-0017.md

```diff
@@ -8,105 +8,465 @@ Category: Interface
+    }
+}
+
 ```
 
-- The params can be found in the [kadena-signing-api](https://kadena-io.github.io/signing-api/). Future methods may be added and will be documented there
+#### kadena_quickSign_v1 Request
 
-## Events
+This method needs to be called with a `params` object with the following properties:
 
-- At the moment we see no need to standardize any events and believe that most usecases can be handled at the kadena-protocol level that sits atop wallet connect. This is subject to change in the future as we understand the workflows more through extended use of the protocol
+1. `params`: `Object`\
+    1.1. `commandSigDatas`: `Array<Object>` - A list of `CommandSigData` to sign. The properties of this type are defined in
+[kip-0015 QuickSign Signing API v1](https://github.com/kadena-io/KIPs/blob/master/kip-0015.md#commandsigdata).
 
-## SLIP-0044
-While not required in the namespace proposal process, a network's SLIP-0044 value is used throughout the wallet connect demo code.
-Kadena's coin type is 626 as documented in https://github.com/satoshilabs/slips/blob/master/slip-0044.md
 
-# Example Proposal and Response
+#### kadena_quickSign_v1 Response
+
+The method response contains a `result` object that contains the following properties if signing was successful:
+
+1. `result`: `Object`\
+    1.1. `responses`: `Array<Object>` - A list of `Response` contained signed . The properties of this type are defined in
+[kip-0015 QuickSign Signing API v1](https://github.com/kadena-io/KIPs/blob/master/kip-0015.md#response-1).
+
+If signing was NOT successful, then the `result` object contains the following properties:
+1. `result`: `Object`\
+    1.1. `error`: `Object` - A list of `QuicksignError`. The properties of this type are defined in
+[kip-0015 QuickSign Signing API v1](https://github.com/kadena-io/KIPs/blob/master/kip-0015.md#quicksignerror).
+
+### kadena_sign_v1
+
+Sign is part of the [Kadena Signing API](https://github.com/kadena-io/signing-api). This method allows the wallet to show the user a **single** transaction that needs signature approval.
+
+#### Example kadena_sign_v1 Request and Response
 
-```json
-# Proposal namespace request
+```jsonc
+// kadena_sign_v1 Method Request
 {
-  "kadena": {
-    "chains": ["kadena:mainnet01", "kadena:testnet04", "kadena:development"],
-    "methods": ["kadena_quicksign_v1"],
-    "events": []
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "kadena_sign_v1",
+  "params": {
+    signingCmd // Chainweaver type
   }
 }
+
 ```
 
-```json
-# Session namespace response
+```jsonc
+// kadena_sign_v1 Response
 {
-  "kadena": {
-    "accounts": [
-      "kadena:mainnet01:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
-      "kadena:testnet04:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
-      "kadena:testnet04:22ddc64851718e9d41d98b0f33d5e328ae5bbbbd97aed9885adac0f2d070ff9c"
-    ],
-    "methods": ["kadena_quicksign_v1"],
-    "events": []
+  "signedCmd": {
```

Comment:
Is this data structure (the Response type of /sign) defined anywhere? I see it defined at https://kadena-io.github.io/signing-api/#/definitions/SigningResponse, but it doesn't match the properties specified in the KIP

Reply:
The Request type in the Haskell code (https://github.com/kadena-io/signing-api/blob/master/kadena-signing-api/src/Kadena/SigningApi.hs#L41) matches the definition in the Open API spec (https://kadena-io.github.io/signing-api/#/definitions/SigningRequest).

The Response type in the Haskell code (https://github.com/kadena-io/signing-api/blob/master/kadena-signing-api/src/Kadena/SigningApi.hs#L60) matches the definition in the Open API spec (https://kadena-io.github.io/signing-api/#/definitions/SigningResponse).

I think some of the discrepancy we're seeing is that pact-lang-api's `signWallet` function (https://github.com/kadena-io/pact-lang-api/blob/fd04d695e6a4ea2203308f25fc781185e0bb3e0a/pact-lang-api.js#L785) formats the `signingCmd` type to match the schema defined above. 

Note the mapping between the `SigningCmd` type in pact-lang-api and what Chainweaver expects (https://github.com/kadena-io/pact-lang-api/blob/fd04d695e6a4ea2203308f25fc781185e0bb3e0a/pact-lang-api.js#L813).

Also, the Response schema has Chainweaver returning the fields `body` and `chainId`. But the `signWallet` function returns just the contents of the `body` field (See https://github.com/kadena-io/pact-lang-api/blob/fd04d695e6a4ea2203308f25fc781185e0bb3e0a/pact-lang-api.js#L788).